mpqp/execution/result.py

```diff
@@ -91,6 +91,37 @@ def __str__(self):
     def __repr__(self) -> str:
         return f"StateVector({clean_1D_array(self.vector)})"
 
+    def to_dict(self) -> dict[str, list[str] | Any | int]:
+        """
+        Converts the StateVector object into a dictionary format.
```

Comment:
the doc for various `to_dict` functions are inhomogeneous, this being said, if
you make them into a `__dict__`, it doesn't matter because we could remove the
doc then